File: src/main/java/net/golem/raknet/protocol/connected/reply/OpenConnectionReply1Packet.java
package net.golem.raknet.protocol.connected.reply;

import net.golem.raknet.codec.PacketDecoder;
import net.golem.raknet.codec.PacketEncoder;
import net.golem.raknet.protocol.DataPacket;
import net.golem.raknet.protocol.RakNetPacketIds;

public class OpenConnectionReply1Packet extends DataPacket {

	public long guid;

	public boolean useSecurity = false;

	public short maximumTransferUnits;

	public OpenConnectionReply1Packet() {
		super(RakNetPacketIds.OPEN_CONNECTION_REPLY_1);
	}

	@Override
	public void encode(PacketEncoder encoder) {
		encoder.writeMagic();
		encoder.writeLong(guid);
		encoder.writeBoolean(useSecurity);
		encoder.writeShort(maximumTransferUnits);
	}

	@Override
	public void decode(PacketDecoder decoder) {
		decoder.skipMagic();
		guid = decoder.readLong();
		useSecurity = decoder.readBoolean();
		maximumTransferUnits = decoder.readShort();
	}
}

File: src/main/kotlin/raknet/packet/protocol/connected/reply/OpenConnectionReply1Packet.kt
package raknet.packet.protocol.connected.reply

import io.netty.buffer.ByteBuf
import raknet.Magic
import raknet.Magic.readMagic
import raknet.packet.DataPacket
import raknet.packet.PacketType

class OpenConnectionReply1Packet(
    var magic: Magic,
    var serverGuid: Long,
    var useSecurity: Boolean,
    var mtuSize: Int
): DataPacket(PacketType.OPEN_CONNECTION_REPLY_1.id()) {


    override fun encodeOrder(): Array<Any> {
        return arrayOf(magic, serverGuid, useSecurity, mtuSize)
    }

    companion object {

        fun from(buffer: ByteBuf): OpenConnectionReply1Packet {
            return OpenConnectionReply1Packet(
                buffer.readMagic(),
                buffer.readLong(),
                buffer.readBoolean(),
                buffer.readInt()
            )
        }
    }

    override fun toString(): String {
        return "OpenConnectionReply1Packet(magic=$magic, serverGuid=$serverGuid, useSecurity=$useSecurity, mtuSize=$mtuSize)"
    }
}